### PR TITLE
[RDY] Refactor vip

### PR DESCRIPTION
**Pull request by Alberth289346**

### CODE REVIEW

File: CorsixTH/Lua/entities/vip.lua [hunk 1]
```diff
@@ -255,6 +255,22 @@ function Vip:evaluateEmergency(success)
...
+--! the provided value is at or below the upper bound.
+--!return Evaluation value.
+local function evaluateProperty(value, eval_table)
+  for _, boundary in ipairs(eval_table) do
+    if not boundary.upper or boundary.upper >= value then
+      return boundary.value
+    end
+  end
+  -- Should never get here, but just in case
+  return 0
```
Comment (TheCycoONE) [inline CorsixTH/Lua/entities/vip.lua]: Should assert or print paranoia debug output here, don't want exception cases to be silent.

File: CorsixTH/Lua/entities/vip.lua [hunk 2]
```diff
@@ -255,6 +255,22 @@ function Vip:evaluateEmergency(success)
   end
 end
 
+--! Evaluate a value based on an evaluation array.
+--!param value (number) Value to evaluate.
+--!param eval_table (table of {upper, value}) Evaluation array with increasing
+--! upper bound (leave out for the final entry), and the value to return if
+--! the provided value is at or below the upper bound.
+--!return Evaluation value.
+local function evaluateProperty(value, eval_table)
```
Comment (TheCycoONE) [inline CorsixTH/Lua/entities/vip.lua]: name isn't great. Maybe rangeMapLookup?
Could possibly be general enough (or generalized enough) to move up to utilities.

File: CorsixTH/Lua/entities/vip.lua [hunk 3]
```diff
@@ -265,15 +281,14 @@ function Vip:setVIPRating()
...
-    if deathRatio > 12 then
-      self.vip_rating = self.vip_rating + 10
-    elseif deathRatio >= 8 then
-      self.vip_rating = self.vip_rating + 5
-    elseif deathRatio >= 2 and deathRatio <= 4 then
-      self.vip_rating = self.vip_rating - 10
-    elseif deathRatio < 2 then
-      self.vip_rating = self.vip_rating - 20
-    end
+    local death_ratio_table = {
```
Comment (TheCycoONE) [inline CorsixTH/Lua/entities/vip.lua]: If we agree to call this a range map above I'd like to keep that name here - e.g. death_ratio_rangemap.

**Comment by TheCycoONE** [comment]
:+1: from sight inspection of all the averages changes.

**Comment by Alberth289346** [comment]
Generalized the rangemap function, and moved it to utils; merged all the averaging changes into the 2nd commit. Aside from 3 additional comments "default value does not generate a message", it's equivalent.
Added new commit to rename camel case variables to lower_underscore_case to unify the variable names somewhat. Tested with a level 1 save, with a VIP visit, giving a report.

**Pull request merged**